design-toolkit/components/src/providers/theme-provider.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use client';
+
+import 'client-only';
+import type { ColorTokens, ThemeTokens } from '@/tokens/types';
+import { merge } from 'lodash';
+import {
+  type PropsWithChildren,
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import type { PartialDeep } from 'type-fest';
+import { designTokens } from '../tokens/tokens';
+
+type Mode = 'dark' | 'light';
+
+type UseThemeContext = {
+  mode: Mode;
+  tokens: ColorTokens;
+  toggleMode: (mode: Mode) => void;
+} | null;
```

Comment:
i was thinking it might be nice to provide default values instead of null for the initial context. then the apps don't need to worry about null checks/optional chaining.

wdyt?